### PR TITLE
Add ns in sdk

### PR DESCRIPTION
**Pull request by gaoning777**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2665)

This PR includes the changes in #2651.
We can first merge the other one before reviewing this.

### CODE REVIEW

File: sdk/python/kfp/_client.py [hunk 1]
```diff
@@ -294,7 +294,7 @@ def list_pipelines(self, page_token='', page_size=10, sort_by=''):
     """
     return self._pipelines_api.list_pipelines(page_token=page_token, page_size=page_size, sort_by=sort_by)
 
-  def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, params={}, pipeline_id=None):
+  def run_pipeline(self, experiment_id, job_name, pipeline_package_path=None, params={}, pipeline_id=None, namespace=None):
```
Comment (hongye-sun) [inline sdk/python/kfp/_client.py]: You probably need to update the doc comment about the new arg and what's the default value if it's not provided.

Comment (hongye-sun) [inline sdk/python/kfp/_client.py]: There are other functions that can run a pipeline. Do you want to add namespace to the other ones? like: create_run_from_pipeline_func

Comment (hongye-sun) [inline sdk/python/kfp/_client.py]: Have you considered to add namespace as part of the client's setting or from env? The namespace is similar with the project id concept in gcloud or gcp sdk.

Comment (gaoning777) [reply sdk/python/kfp/_client.py]: Done.

Comment (gaoning777) [reply sdk/python/kfp/_client.py]: Done.

Comment (gaoning777) [reply sdk/python/kfp/_client.py]: Synced offline. Added TODOs.

**Comment by IronPan** [comment]
Thanks Ning!

**Pull request merged**